spec/support/docker/servicenow/mock_instance.rb

```diff
@@ -74,6 +74,10 @@ class MockServiceNowInstance < Sinatra::Base
     nil
   end
 
+  get '/healthcheck' do
```

Comment:
Hmm, is this an actual ServiceNow API endpoint? If not, we should probably remove it. Remember, the tests _should_ work with an actual ServiceNow instance provided that instance satisfies some preconditions (currently the only precondition is that it has the `Puppet Classes` and `Puppet Environment` fields defined so that classification tests can pass).

Reply:
This endpoint isn't for the tests. It's for the healthcheck in the `docker-compose.yaml`. Without this endpoint it's more difficult to write a health check that knows when the server is running and actually ready to serve requests. Since the tests never reference it, I disagree that it should be removed.

Reply:
Ah I see you put the ServiceNow container in `docker-compose`. I don't think we should do that -- I think `docker-compose` should be interpreted as "This is everything you need to setup a containerized PE environment". Reason is b/c I should still be able to get these tests working _if_ I wanted to use a real ServiceNow instance on a containerized PE environment.

I think `setup_servicenow_instance` could have some code under https://github.com/puppetlabs/puppetlabs-servicenow_cmdb_integration/blob/master/Rakefile#L127 that's something like

```
if using a containerized PE environment
  // we start mock instance on localhost
else
 // we start mock instance on the master VM
end
```

Reply:
I don't really want to deal with wrinkle right now. If and when we actually do start testing against an external servicenow instance we can factor this out. For right now running the servicenow container as a separate docker container not attached to the compose environment is not as simple as it sounds. You can have the puppet container in the compose environment refer to the servicenow contain by `localhost` because inside the container `localhost` doesn't refer to the host machine you're running the container on. It refers to the container itself, and that's not where the servicenow container is running. Getting a container to refer to ports on the host machine is not idiomatic, and getting it to happen is hacky and weird. The best you can do is create the servicenow container after the compose environment is fully up and running, and run another command that attaches it to the compose environments internal network. For now I just want to be able to run `docker-compose up` and have everything I need ready.

Reply:
Actually I think things should still work -- I can still do `setup_servicenow_instance[fqdn,user,password]` and all that does is change `inventory.yaml` file to an actual ServiceNow instance. So worst-case, we'll just have an extra ServiceNow container running that isn't actually being used.

So, this isn't that big of a deal. If putting the container in docker-compose makes it easier for the PE environment to reference it, then 👍 . I'd maybe add a note about that somewhere, probably in the README.